app/lib/dfe_data_tables/categories_parser.rb

```diff
@@ -42,17 +42,21 @@ def categories
 
         concept_hash = concept(row)
 
-        if row[3].present?
+        l0 = obj&.last
+        l1 = l0&.dig(:subcat)&.last
+        l2 = l1&.dig(:subcat)&.last
+        l3 = l2&.dig(:subcat)&.last
+        unless l3.nil?
```

Comment:
Could we name these variables meaningfully to make the intent clearer?

Reply:
Done